fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/values/RecordConstructorValue.java

```diff
@@ -387,8 +387,7 @@ public static RecordConstructorValue fromProto(@Nonnull final PlanSerializationC
             columnsBuilder.add(Column.fromProto(serializationContext, columnProto));
         }
         final ImmutableList<Column<? extends Value>> columns = columnsBuilder.build();
-        Verify.verify(!columns.isEmpty());
```

Comment:
Yeah. In particular, this comes up because `count` queries use an empty record to represent the grouped value in their `GroupByExpression`. This came up in the context of this PR because I added a `verifySerialization` with a count query, and the plan serialization failed.  Similar story here: https://github.com/FoundationDB/fdb-record-layer/pull/2473/files#r1478698348

As far as I can tell, serialization of a plan using a count index _always_ failed (this isn't from something I added, other than the verification call).